denonavr/foundation.py

```diff
@@ -622,6 +648,18 @@ async def async_options(self) -> None:
 
     async def async_settings_menu(self) -> None:
         """Options menu on receiver via HTTP get command."""
+        # fast path if we already know the state
+        if self.telnet_available and self._settings_menu is not None:
```

Comment:
```suggestion
        if self.telnet_available:
```
This looks for me like the telnet path instead of the fast path.
You could add a `MNMEN?` command [here](https://github.com/ol-iver/denonavr/blob/0e4f0338a6865759db7c9b110a2e59a1e1fb0dd9/denonavr/api.py#L569) which make the library query the state each time telnet connects.

Then, you could also remove the telnet handling in the slow path.

Reply:
I hope I understood you correctly with my adjustments :)